src/main/java/io/leonis/subra/ipc/peripheral/JamepadController.java

```diff
@@ -1,14 +1,21 @@
 package io.leonis.subra.ipc.peripheral;
 
 import com.studiohartman.jamepad.ControllerState;
+import io.leonis.subra.game.data.Player;
+import io.leonis.zosma.game.Identity;
 import io.leonis.zosma.ipc.peripheral.Controller;
 import lombok.Value;
 
 /**
  * @author Jeroen de Jong
  */
 @Value
-public class JamepadController implements Controller<Integer, ControllerState> {
-  private final Integer identifier;
+public class JamepadController implements Controller<JamepadController.Identity, ControllerState> {
+  private final Identity identifier;
   private final ControllerState controls;
+
+  @Value
+  public static class Identity implements io.leonis.zosma.game.Identity {
```

Comment:
I think it would be clearer if this was renamed to `JamepadControllerIdentity` or something (`JamepadIdentity` would suffice as well)